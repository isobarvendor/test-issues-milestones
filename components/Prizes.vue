<template>
  <div class="container campaign-section" v-if="data">
    <div class="header">{{data.title}}</div>
    <div v-html="data.description"></div>
    <div class="container">
      <swiper class="swiper prize-swiper" :options="swiperOption">
        <swiper-slide
          v-for="(item, index) in data.exclusizePrizesSection"
          :key="'prize'+index"
          class="prize-item"
        >
          <img
            v-if="$mq == 'sm'"
            :src="item.image ? item.image[0].url : '/img/prize-default.svg'"
          />
          <img v-else :src="item.image? item.image[0].url : '/img/prize-default.svg'" />
          <div class="prize-description">{{item.title}}</div>
        </swiper-slide>
        <div class="prize-swiper-pagination" slot="pagination"></div>
      </swiper>
    </div>
    <div v-if="winners.luckyWinnerSection" class="winners-section">
      <div class="header">{{winners.title}}</div>
      <a class="button" href="/winners">{{winners.ButtonTitle}}</a>
    </div>
  </div>
</template>

<script>
import { Swiper, SwiperSlide } from "vue-awesome-swiper";
export default {
  components: {
    Swiper,
    SwiperSlide,
  },
  data() {
    return {
      swiperOption: {
        slidesPerView: 2,
        spaceBetween: 10,
        breakpoints: {
          // when window width is >= 320px
          768: {
            slidesPerView: 3,
            spaceBetween: 20,
          },
          992: {
            slidesPerView: 3,
            spaceBetween: 30,
          },
        },
        freeMode: true,
        pagination: {
          el: ".prize-swiper-pagination",
          clickable: true,
          bulletClass: "prize-swiper-pagination-bullet",
          bulletActiveClass: "prize-swiper-pagination-bullet-active",
        },
      },
    };
  },
  props: {
    data: null,
    winners: null,
  },
  beforeMount() {},
};
</script>
<style scoped lang="scss">
.campaign-section{
.winners-section{
  .button{
    color: #333333;
    background: #ffffff;
    font-size: 14px;
    width: 370px;
    margin:auto;
    display: table;
    margin-top: 35px;
    padding: 20px 90px;
  }
}
@media only screen and (max-width: 1199px) {
  .winners-section{
    .button{
      width:315px;
    }
  }
}
}
</style>
