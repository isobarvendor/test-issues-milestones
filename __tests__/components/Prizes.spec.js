
import { shallowMount } from '@vue/test-utils'

import Prizes from "@/components/Prizes";


const factory = () => {
  return shallowMount(Prizes, {
    propsData: {
      data:{
        exclusizePrizesSection:[
          {
            image:[
              {url: "imageurl"}
            ],
            title:"Exclusive Prize"
          }
        ],
        title:"title",
        description:"description"
      },
      winners:{
        luckyWinnerSection:[{
          lucky:"tes"
        }],
        title:"title",
        ButtonTitle:"Button"
      },
    },
  });
};

describe("Prizes", () => {


  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
