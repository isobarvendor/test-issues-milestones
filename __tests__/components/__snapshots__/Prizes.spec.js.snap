// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prizes renders properly 1`] = `
<div class="container campaign-section" winnners="[object Object]">
  <div class="header">title</div>
  <div>description</div>
  <div class="container">
    <swiper-stub defaultoptions="[object Object]" options="[object Object]" autoupdate="true" autodestroy="true" deleteinstanceondestroy="true" cleanupstylesondestroy="true" class="swiper prize-swiper">
      <swiper-slide-stub class="prize-item"><img src="imageurl">
        <div class="prize-description">Exclusive Prize</div>
      </swiper-slide-stub>
      <div class="prize-swiper-pagination"></div>
    </swiper-stub>
  </div>
  <!---->
</div>
`;
